emr-common/src/main/scala/com/aliyun/emr/common/DynamicConfigManager.scala

```diff
@@ -27,48 +27,37 @@ import org.apache.curator.framework.{CuratorFramework, CuratorFrameworkFactory}
 import org.apache.curator.framework.recipes.cache.{PathChildrenCache, PathChildrenCacheEvent, PathChildrenCacheListener}
 import org.apache.curator.retry.BoundedExponentialBackoffRetry
 
-class DynamicConfigManager(checkpointRoot: String, zkConnect: String, sessionTimeoutMs: Int) {
+class DynamicConfigManager(rootPath: String, zkConnect: String, sessionTimeoutMs: Int) {
   private val curator: CuratorFramework = CuratorFrameworkFactory.newClient(
     zkConnect, new BoundedExponentialBackoffRetry(100, 100, 20))
   curator.start()
 
-  private val zNodeChangeHandlers = new ConcurrentHashMap[String, ZNodeChangeHandler]().asScala
-  private val zNodeChildChangeHandlers = new ConcurrentHashMap[String, ZNodeChildChangeHandler]().asScala
+  private val zNodeChangeHandlers = new ConcurrentHashMap[(String, String), ZNodeChangeHandler]().asScala
```

Comment:
Different loghub source in one job use the same checkpoint path.  Use `(checkpointPath, source name)` to replace `checkpointPath`